### PR TITLE
chore: Update list of owners and maintainers

### PR DESCRIPTION
**Pull request by rm3l**
## Description
This simply cleans up the list of owners and maintainers, to reduce the number of review requests automatically sent out to people no longer actively working on the project.

## Which issue(s) does this PR fix or relate to

&mdash;

## PR acceptance criteria

- [ ] Tests
- [ ] Documentation
- [x] If the bundle manifests have been updated, make sure to review the [`rhdh-operator.csv.yaml`](../.rhdh/bundle/manifests/rhdh-operator.csv.yaml) file accordingly

## How to test changes / Special notes to the reviewer
&mdash;

### CODE REVIEW

File: bundle/manifests/backstage-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -300,6 +300,8 @@ spec:
   maintainers:
   - email: asoro@redhat.com
     name: Armel Soro
+  - email: cdaley@redhat.com
+    name: Corey Daley
```
Comment (nickboldt) [inline bundle/manifests/backstage-operator.clusterserviceversion.yaml:304]: if you're adding Corey and removing Jianrong in the pr build, you could also remove Jianrong here.

File: config/manifests/bases/backstage-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -44,6 +44,8 @@ spec:
   maintainers:
   - email: asoro@redhat.com
     name: Armel Soro
+  - email: cdaley@redhat.com
+    name: Corey Daley
```
Comment (nickboldt) [inline config/manifests/bases/backstage-operator.clusterserviceversion.yaml:48]: if you're adding Corey and removing Jianrong in the pr build, you could also remove Jianrong here.

File: .rhdh/bundle/manifests/rhdh-operator.csv.yaml [hunk 3]
```diff
@@ -338,6 +338,8 @@ spec:
   maintainers:
   - email: asoro@redhat.com
     name: Armel Soro
+  - email: cdaley@redhat.com
+    name: Corey Daley
```
[the remark this answers is not in the document]
Comment (rm3l) [reply .rhdh/bundle/manifests/rhdh-operator.csv.yaml:342]: Good catch. Fixed in [`30b044b` (#364)](https://github.com/janus-idp/operator/pull/364/commits/30b044b07d8b9b74bc58cccc034b28f0ca3c0ce0)

File: bundle/manifests/backstage-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (rm3l) [reply bundle/manifests/backstage-operator.clusterserviceversion.yaml:304]: Good catch. Fixed in [`30b044b` (#364)](https://github.com/janus-idp/operator/pull/364/commits/30b044b07d8b9b74bc58cccc034b28f0ca3c0ce0)

File: config/manifests/bases/backstage-operator.clusterserviceversion.yaml (hunk 2 above)
Comment (rm3l) [reply config/manifests/bases/backstage-operator.clusterserviceversion.yaml:48]: Good catch. Fixed in [`30b044b` (#364)](https://github.com/janus-idp/operator/pull/364/commits/30b044b07d8b9b74bc58cccc034b28f0ca3c0ce0)

**Review by nickboldt — APPROVED** [review]

**Pull request merged**